src/Microsoft.Framework.Configuration/ConfigurationBase.cs

```diff
@@ -86,12 +86,20 @@ public IConfigurationSection GetSection(string key)
         {
             if (string.IsNullOrEmpty(key))
             {
-                throw new InvalidOperationException(Resources.Error_EmptyKey);
+                throw new ArgumentException(Resources.Error_EmptyKey);
```

Comment:
I would blow this up. Key will get validated in the constructor of `ConfigurationSection` at most this is one method call frame in the stack.